tests/application_development/ram_context_for_isr/include/fake_driver.h

```diff
@@ -11,19 +11,18 @@
 extern "C" {
 #endif
 
-#ifdef CONFIG_BOARD_QEMU_CORTEX_M3
-
-#define TEST_IRQ_NUM  42
-#define TEST_IRQ_PRIO 1
+#define DT_DRV_COMPAT fakedriver
 
+#if DT_NODE_HAS_PROP(DT_DRV_INST(0), interrupts)
+#define TEST_IRQ_NUM	DT_IRQN(DT_DRV_INST(0))
+#define TEST_IRQ_PRIO	DT_IRQ(DT_DRV_INST(0), priority)
```

Comment:
```suggestion
#define TEST_IRQ_NUM	DT_INST_IRQN(0)
#define TEST_IRQ_PRIO	DT_INST_IRQ(0, priority)
```

\+ Nit: could use a comment
```c
/* Platforms with specific needs provide IRQn
 * and priority via DT - use that if available.
 */
```